### ISSUE TITLE
Self intersections/other geometry issues in flowlines

### ISSUE THREAD

**Issue by ChristopherTracey** [HeritageNetwork/Regional_SDM#119]
In running some test models, @memoore and I got the following error in some models run in the Susquehanna/Delaware basin:
```
Error in CPL_geos_union(st_geometry(x), by_feature) : 
  Evaluation error: TopologyException: Input geom 1 is invalid: Self-intersection at or near point 1674622 1982562 at 1674622 1982562.
```
This is in the predict step when its trying to join the probabilities to the flowlines. My guess is its a bad WKT string.  Any ideas on how to track it down?

**Comment by memoore** [comment]
I'm also getting the following error for species in the Ohio basin at the same step:
`Error in CPL_geos_union(st_geometry(x), by_feature)
    Evaluation error: TopologyException: Input geom 1 is invalid: Hole lies outside shell at or near point 598828.30000000005 2898123 at 598828.30000000005 2898123.`

**Comment by ChristopherTracey** [comment]
This thread has some guidelines on checking feature geometries using `sf`: https://www.r-spatial.org/r/2017/03/19/invalid.html and specific mentions self-intersections and holes.

I'm thinking of adding these checks after the conversion from wkt. Not sure though...

**Comment by ChristopherTracey** [comment]
This issue actually relates to HUC12s and not the flowlines. The original error was interpreted to happen on the flowline step, but running the code line by line showed that it was clear.  We're rebuilding the geometry on the HUC12s.

**Comment by ChristopherTracey** [comment]
Fixed in #127 by adding a zero-width buffer to the selected HUC12s.  Other methods, including trying to fix in Arc, did not work.

@memoore, told you I could fix this by the weekend.
